lib/task/import/validate/csvColumnNameValidator.class.php

```diff
@@ -0,0 +1,146 @@
+            foreach ($header as $columnName) {
+                // If $columnName is not in validColumnNames.
+                if (!in_array($columnName, $this->validColumnNames)) {
+                    // Test for leading or trailing whitespace.
+                    if (in_array(trim($columnName), $this->validColumnNames)) {
+                        $this->trimIssuesColumnNames[$columnName] = trim($columnName);
+                    }
+                    // Test if a match is found using lowercase trimmed matching.
+                    elseif (false !== $key = array_search(strtolower(trim($columnName)), $this->validColumnNamesLowercase)) {
+                        // Map unknown column name to possible match.
+                        $this->caseIssuesColumnNameMap[trim($columnName)] = $this->validColumnNames[$key];
+                    }
+
+                    // Add to unknown column name list.
+                    $this->unknownColumnNames[$columnName] = $columnName;
+                }
+            }
+
+            $this->complete = true;
+        }
+    }
+
+    public function getTestResult()
+    {
+        $this->testData->addResult(sprintf('Number of unrecognized column names found in CSV: %s', count($this->unknownColumnNames)));
+
+        if (0 < count($this->unknownColumnNames)) {
+            $this->testData->setStatusWarn();
+            $this->testData->addResult('Unrecognized columns will be ignored by AtoM when the CSV is imported.');
+            foreach ($this->unknownColumnNames as $unknownColumnName) {
+                $this->testData->addDetail(sprintf('Unrecognized column: %s', $unknownColumnName));
+            }
+        }
+
+        if (0 < count($this->trimIssuesColumnNames)) {
+            $this->testData->setStatusWarn();
+            $this->testData->addResult(sprintf('Number of column names with leading or trailing whitespace characters: %s', count($this->trimIssuesColumnNames)));
+            $this->testData->addDetail(sprintf('Column names with leading or trailing whitespace: %s', implode(',', $this->trimIssuesColumnNames)));
+        }
+
+        if (0 < count($this->caseIssuesColumnNameMap)) {
+            $this->testData->setStatusWarn();
+            $this->testData->addResult(sprintf('Number of unrecognized columns that may be case related: %s', count($this->caseIssuesColumnNameMap)));
+            foreach ($this->caseIssuesColumnNameMap as $key => $value) {
+                $this->testData->addDetail(sprintf('Possible match for %s: %s', $key, $value));
+            }
+        }
+
+        return parent::getTestResult();
+    }
+
+    // Load the column name config from lib/flatfile/config.
+    protected function loadObjectColumnNames($resourceClass)
+    {
+        $resourceTypeBaseConfigFile = $resourceClass.'.yml';
+        $config = QubitFlatfileExport::loadResourceConfigFile($resourceTypeBaseConfigFile, 'base');
+
+        $this->validColumnNames = $config['columnNames'];
```

Comment:
I think this will cause a PHP warning if 'columnNames' isn't defined in the config file?

Reply:
Good catch. I assumed that each yml file would _always_ contain a 'columnNames' section. Will check this out....

Reply:
The logic that I've used to parse these files will fallback to assembling the valid columns from other sections in this file. As it is, all existing entity validation files do contain a 'columnNames' section. I have added a new commit that will throw an exception if the yml file for the entity in question is missing entirely.